File: app/src/main/java/com/dynamo/spacex/viewmodels/SpaceXViewModel.java
package com.dynamo.spacex.viewmodels;

import androidx.databinding.ObservableField;
import androidx.lifecycle.ViewModel;

import com.dynamo.spacex.model.PastLaunch;

import javax.inject.Inject;

public class SpaceXViewModel extends ViewModel {

    public ObservableField<String> missionName = new ObservableField<>();
    public ObservableField<String> missionTimestamp = new ObservableField<>();

    private PastLaunch pastLaunch;

    @Inject
    public SpaceXViewModel() { }

    public PastLaunch getPastLaunch() {
        return this.pastLaunch;
    }

    void setPastLaunch(PastLaunch pastLaunch) {
        this.pastLaunch = pastLaunch;
    }
}

File: app/src/main/java/com/dynamo/spacex/viewmodels/SpaceXViewModel.kt
package com.dynamo.spacex.viewmodels

import androidx.databinding.ObservableField
import com.dynamo.spacex.model.PastLaunch

class SpaceXViewModel {
    val missionName: ObservableField<String> = ObservableField()
    val missionTimestamp: ObservableField<String> = ObservableField()
    lateinit var pastLaunch: PastLaunch
}